File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo): # 사람 이름 / 사람별 그리움 점수 / 찍힌 인물이름
    answer = []
    dict = {}
    for i ,n in enumerate(name):
        dict[n]= yearning[i] 
    for i in range(len(photo)):
        ans = 0
        for j in range(len(photo[i])):
            if dict.get(photo[i][j],0)!=0:
                ans += dict[photo[i][j]]
        answer.append(ans)
    return answer

File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.cpp
#include <string>
#include <vector>
#include <map>
#include <iostream>
using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) { //그리워하는 사람의 이름, 그리움 점수, 사진에 찍힌 인물의 이름 
    vector<int> answer;
    map<string,int> score;
    for(int i=0;i < name.size() ; i++){
        score[name[i]] = yearning[i];
    }
    for(int i = 0 ; i< photo.size();i++){
        int s = 0;
        for (int j =0 ; j< photo[i].size();j++){
            s += score[photo[i][j]];
        }
        answer.push_back(s);
    }
    return answer;
}